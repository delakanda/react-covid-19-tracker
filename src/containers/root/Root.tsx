import React from 'react';
import { Provider } from 'react-redux';
import App from '../app/App';
import Store from './../../redux/store/Store';
import 'bootstrap/dist/css/bootstrap.min.css';

function Root() {
  return (
    <Provider store={Store}>
      <App />
    </Provider>
  );
}

export default Root;
