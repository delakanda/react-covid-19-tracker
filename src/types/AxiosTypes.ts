import { Method } from "axios";

export type AxiosCallParams = {
  method: Method;
  url: string;
  params?: object;
  data?: any;
  headers?: object;
};