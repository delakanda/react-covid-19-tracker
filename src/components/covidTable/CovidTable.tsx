import React from 'react'
import { formatNum } from '../../utils/NumberUtil';
import { ICovidSummaryData } from '../../types/CovidTypes';

type TCovidTableProps = {
  covidSummaryData: ICovidSummaryData | null;
};

function CovidTable(props: TCovidTableProps) {
  return (
    <table className="table table-striped">
      <tbody>
        <tr>
          <th>Confirmed</th>
          <th>{formatNum(props.covidSummaryData?.total_cases) || "--"}</th>
        </tr>
        <tr>
          <th>Active Cases</th>
          <th>{formatNum(props.covidSummaryData?.active_cases) || "--"}</th>
        </tr>
        <tr>
          <th>Recoveries</th>
          <th>{formatNum(props.covidSummaryData?.recovered) || "--"}</th>
        </tr>
        <tr>
          <th>Deaths</th>
          <th>{formatNum(props.covidSummaryData?.deaths) || "--"}</th>
        </tr>
      </tbody>
    </table>
  );
}

export default CovidTable;