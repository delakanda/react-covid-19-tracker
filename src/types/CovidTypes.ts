import { FetchState } from "./Default";

export type TCovidReducer = {
  countrySelection: string;
  fetchState: FetchState;
  immutableCovidSummaryData: ICovidSummaryData | null;
  covidSummaryData: ICovidSummaryData | null;
  covidCountriesData: ICovidCountryData[];
  lastUpdated: string | null;
};

export interface ICovidSummaryData {
  total_cases: number;
  active_cases: number;
  deaths: number;
  recovered: number;
  death_ratio: number;
  recovery_ratio: number;
};

export interface ICovidCountryData extends ICovidSummaryData {
  name: string;
};