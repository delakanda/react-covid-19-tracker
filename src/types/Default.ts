import { TCovidReducer } from "./CovidTypes";

export enum FetchState {
  Fetching = "Fetching",
  Fetched = "Fetched",
  FetchError = "FetchError",
  FetchNeutral = "FetchNeutral"
};

export type TReduxReducers = {
  covidReducer: TCovidReducer;
}

export type TReducerPayload = {
  type: string;
  payload?: any
};