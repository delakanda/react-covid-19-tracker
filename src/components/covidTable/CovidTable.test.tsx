import React from 'react';
import CovidTable from './CovidTable';
import { shallow, mount } from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Root from '../../containers/root/Root';
import { ICovidSummaryData } from '../../types/CovidTypes';
import renderer from 'react-test-renderer';

Enzyme.configure({ adapter: new Adapter() });


describe('App Component', () => {

  let mockData: ICovidSummaryData = {
    active_cases: 100,
    death_ratio: 0.1,
    deaths: 100,
    recovered: 100,
    recovery_ratio: 0.1,
    total_cases: 100
  };
  
  let wrapper: Enzyme.ShallowWrapper;

  beforeEach(() => {
    wrapper = shallow(<CovidTable covidSummaryData={mockData} />);
  });

  it("CovidTable Renders", () => {
    mount(<CovidTable covidSummaryData={mockData} />);
  });

  it('Verify Table Row Count', () => {
    const tableRows = wrapper.find('tr');
    expect(tableRows.length).toBe(4);
  });

  it('Verify Snapshot', () => {
    const tree = renderer
      .create(<CovidTable covidSummaryData={mockData} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
