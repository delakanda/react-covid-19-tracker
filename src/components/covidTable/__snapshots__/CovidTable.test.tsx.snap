// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Very Snapshot 1`] = `
<table
  className="table"
>
  <tbody>
    <tr>
      <th
        className="text-info"
      >
        Confirmed
      </th>
      <th
        className="text-info"
      >
        100
      </th>
    </tr>
    <tr>
      <th
        className="text-info"
      >
        Active Cases
      </th>
      <th
        className="text-info"
      >
        100
      </th>
    </tr>
    <tr>
      <th
        className="text-success"
      >
        Recoveries
      </th>
      <th
        className="text-success"
      >
        100
      </th>
    </tr>
    <tr>
      <th
        className="text-danger"
      >
        Deaths
      </th>
      <th
        className="text-danger"
      >
        100
      </th>
    </tr>
  </tbody>
</table>
`;
